pkg/translate/translate_value.go

```diff
@@ -267,26 +263,58 @@ func translateHPASpec(inPath string, outPath string, value interface{}, valueTre
 	if !asEnabled {
 		return nil
 	}
-	newP := util.PathFromString(inPath)[0]
-	minPath := newP + ".autoscaleMin"
-	asMin, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(minPath))
-	if found && err == nil {
-		if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath+".minReplicas"), asMin); err != nil {
+
+	newP := util.PathFromString(inPath)
+	// last path element is k8s setting name
+	newPS := newP[:len(newP)-1].String()
+
+	valMap := map[string]string{
+		".autoscaleMin":                 ".minReplicas",
+		".autoscaleMax":                 ".maxReplicas",
+		".cpu.targetAverageUtilization": ".metrics",
+	}
+
+	for key, newVal := range valMap {
+		valPath := newPS + key
+		asVal, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(valPath))
+		if found && err == nil {
+			log.Infof("path has value in helm Value.yaml tree, mapping to output path %s", outPath)
+			if key == ".cpu.targetAverageUtilization" {
```

Comment:
rather than checking for a special key, it's better to move this outside the loop, then have both this and the loop code call the same helper function if possible.

Reply:
removed the special key out of the loop